src/lib/stitching/exchange/stitching.ts

```diff
@@ -0,0 +1,222 @@
+import { GraphQLSchema } from "graphql"
+import { amountSDL, amount } from "schema/fields/money"
+import gql from "lib/gql"
+
+const orderTotals = [
+  "itemsTotal",
+  "sellerTotal",
+  "commissionFee",
+  "totalListPrice",
+  "buyerTotal",
+  "taxTotal",
+  "shippingTotal",
+  "transactionFee",
+]
+const orderTotalsSDL = orderTotals.map(amountSDL)
+
+const lineItemTotals = ["shippingTotal", "listPrice", "commissionFee"]
+const lineItemTotalsSDL = lineItemTotals.map(amountSDL)
+
+const offerAmountFields = ["amount", "taxTotal", "shippingTotal", "buyerTotal"]
+const offerAmountFieldsSDL = offerAmountFields.map(amountSDL)
+export const exchangeStitchingEnvironment = (
+  localSchema: GraphQLSchema,
+  exchangeSchema: GraphQLSchema
+) => {
+  type DetailsFactoryInput = { from: string; to: string }
+
+  /**
+   * This returns a resolver which can take an exchange OrderParty union
+   * and convert it into a metaphysics `OrderParty`.
+   *
+   * You pass in the field to get the details from, and and the new fieldName
+   */
+  const partyUnionToDetailsFactory = ({ from, to }: DetailsFactoryInput) => {
+    // We abuse the query alias feature to make sure that all
+    // the data we need to generate the full object from
+    // is included.
+
+    // It's possible that this working around a bug.
+    const aliasedPartyFragment = (field, alias) => {
+      return gql`
+      ... on CommerceOrder {
+        ${alias}: ${field} {
+          __typename
+          ... on CommerceUser {
+            __typename
+            id
+          }
+          ... on CommercePartner {
+            __typename
+            id
+          }
+        }
+      }`
+    }
+
+    return {
+      // Bit of a magic in next line, when adding fragment, it seems
+      // all second level fields are ignored, so __typename and id
+      // couldn't be added, so the hack was to alias the fragment field
+      // and that gets the current fields
+      fragment: aliasedPartyFragment(from, to),
+      resolve: (parent, _args, context, info) => {
+        const typename = parent[to].__typename
+        const id = parent[to].id
+
+        // Make a call to the user or partner resolver on query to
+        // grab our Metaphysics representations
+        return (
+          info.mergeInfo
+            .delegateToSchema({
+              schema: localSchema,
+              operation: "query",
+              fieldName: typename === "CommerceUser" ? "user" : "partner",
+              args: {
+                id,
+              },
+              context,
+              info,
+              transforms: (exchangeSchema as any).transforms,
```

Comment:
If we're extending these objects, perhaps they should be a subclass instead and thus removing the need to cast?

Reply:
👍